File: 0020 - Valid Parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> sk;
        for(auto c : s){
            switch(c){
                case '(': sk.push(')'); break;
                case '{': sk.push('}'); break;
                case '[': sk.push(']'); break;
                default:
                    if(sk.empty() || c!=sk.top()) return false;
                    else sk.pop();
            }
        }
        return sk.empty();
    }
};

File: 0020 - Valid Parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack, d = [], {'(':')', '{':'}', '[':']'}
        for i in s:
            if i in d:
                stack.append(d[i])
            elif not stack or i!=stack.pop():
                return False
        return not stack